docs/csharp/language-reference/operators/patterns.md

```diff
@@ -145,7 +145,9 @@ As the preceding example shows, you can repeatedly use the pattern combinators i
 
 ### Precedence and order of checking
 
-The following list orders pattern combinators starting from the highest precedence to the lowest:
+A pattern that is part of an `is` expression has *relational* precedence. It's the same precedence as the `is` operator. Pattern expressions in the `case` arms of a `switch` statement have the same precedence as the *conditional logical* operators (`||` and `&&`). Pattern expressions inside a `switch` expression have the same precedence as the coalescing operators (`??` and `throw` expressions).
```

Comment:
I'm confused. What is precedence when we talk about patterns and operators/expressions combined? I see what precedence means only for operators/expressions. I see what precedence means only for patterns. But what does it mean to say:

> A pattern that is part of an `is` expression has the same precedence as the `is` expression

Examples of how to use this new text would be beneficial and very helpful. Without examples, it looks very spec-like and not practical. 

The example in the linked issue is as follows:

```csharp
var monthBeginsOnWeekend = (DateTime date) =>
	date.DayOfWeek is DayOfWeek.Saturday or DayOfWeek.Sunday
	&& date.Day == 1;
```

There are two items here:

1. How can this example be misinterpreted, precedence-wise? What is a possible confusion here? Why is it worth to create an  issue for it?
2. If there is a confusion with this example that makes it hard to process for somebody, how does the added text help to clarify it?

Also it looks that the new text talks about all the patterns, while the edited section is the subsection of the section about logical patterns.  Should this H3 section be left untouched and another H2 added for the general case, if that's really necessary?

Reply:
To minimize the docs edit, I'm trying to understand what is the origin of the issue. So, coming back to the question in #33397:

> I'm missing the information about how pattern matching operators precedence fits in this table, for example:
>
> what's the priority of `or` and `and`, is it lower or higher than `is`, how about relation to `&&`?

I still don't see the cases where one needs to compare the precedence of `is` with `and`/`or`.

Should we do the minimal edit and specifically clarify the relation between `and`/`or` and `&&`/`||`/`&`/`|`?
That may be important as the following example (can someone come with a *meaningful* similar example) shows:

```csharp
bool a = true;
var c = a is false or true && false;  // (1)
var e = a switch
{
    false or true && false => 0,  // (2)
    _ => 1,
};
```

Case (1) is interpreted as `a is (false or true) && false`, while case (2) is interpreted as `false or (true && false)`

So, would it be enough to only add the following paragraph:

> When a logical pattern is a pattern of an `is` expression, the precedence of logical pattern combinators is **higher** than the precedence of logical operators. Otherwise, the precedence of logical pattern combinators is **lower** than the precedence of logical and conditional logical operators.

I think that addition mostly addresses the original issue, doesn't require the edition of the existing docs (only adds the new paragraph) and clear enough. @BillWagner what do you think?

(Thanks to @jcouv for the clarifications in the original issue.)

Reply:
> So, would it be enough to only add the following paragraph:
> 
> > When a logical pattern is a pattern of an `is` expression, the precedence of logical pattern combinators is **higher** than the precedence of logical operators. Otherwise, the precedence of logical pattern combinators is **lower** than the precedence of logical and conditional logical operators.

This is a great suggestion. I'll take it.